test/jdk/java/awt/color/ICC_Profile/BuiltInProfileCheck/BuiltInProfileCheck.java

```diff
@@ -0,0 +1,165 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8346465
+ * @summary Tests if setData() throws IAE for BuiltIn profiles
+ */
+
+import java.awt.color.ColorSpace;
+import java.awt.color.ICC_Profile;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Map;
+
+public class BuiltInProfileCheck {
+    private static final int HEADER_TAG = ICC_Profile.icSigHead;
+    private static final int INDEX = ICC_Profile.icHdrDeviceClass;
+    private static final String EXCEPTION_MSG = "Built-in profile cannot be modified";
+    private static ICC_Profile testProfile;
+
+    private static final Map<Integer, String> colorSpace = Map.of(
+            ColorSpace.CS_sRGB, "CS_sRGB",
+            ColorSpace.CS_PYCC, "CS_PYCC",
+            ColorSpace.CS_GRAY, "CS_GRAY",
+            ColorSpace.CS_CIEXYZ, "CS_CIEXYZ",
+            ColorSpace.CS_LINEAR_RGB, "CS_LINEAR_RGB"
+    );
+
+    public static void main(String[] args) throws Exception {
+        System.out.println("CASE 1: Testing BuiltIn Profile");
+        for (int cs : colorSpace.keySet()) {
+            prepareTestProfile("Default", true, cs);
+            testProfile(true, cs);
+        }
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 2: Testing Custom Profile");
+        prepareTestProfile("Default", false, ColorSpace.CS_sRGB);
+        testProfile(false, ColorSpace.CS_sRGB);
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 3: Testing Built-In Profile"
+                            + " Serialization & Deserialization");
+        for (int cs : colorSpace.keySet()) {
+            prepareTestProfile("Serialize", true, cs);
+            testProfile(true, cs);
+        }
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 4: Testing Custom Profile"
+                            + " Serialization & Deserialization");
+        prepareTestProfile("Serialize", false, ColorSpace.CS_sRGB);
+        testProfile(false, ColorSpace.CS_sRGB);
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 5: Test reading Built-In profile from .icc file");
+        prepareTestProfile("ReadFromFile", true, ColorSpace.CS_sRGB);
+        testProfile(true, ColorSpace.CS_sRGB);
+        System.out.println("Passed\n");
+
+        System.out.println("CASE 6: Test reading Custom profile from .icc file");
+        prepareTestProfile("ReadFromFile", false, ColorSpace.CS_sRGB);
+        testProfile(false, ColorSpace.CS_sRGB);
+        System.out.println("Passed\n");
+    }
+
+    private static void prepareTestProfile(String testCase,
+                                           boolean isBuiltIn, int cs) {
```

Comment:
I'd probably return the resulting profile to run test on, it will make data flow explicit. However, it'd be more verbose… to pass the test profile.

So, the current approach is sensible.

Could you write a comment that `prepareTestProfile` stores the profile to test in `testProfile`? It could be a javadoc for the `testProfile` field.

Reply:
> I'd probably return the resulting profile to run test on, it will make data flow explicit. However, it'd be more verbose… to pass the test profile.

My initial thought was same - return the profile, but chose to keep it as static var to reduce the number of arguments prepareTestProfile() & testProfile() and it keeps the test code simple.

> Could you write a comment that prepareTestProfile stores the profile to test in testProfile? It could be a javadoc for the testProfile field.

Updated.